pkg/schedule/operator_controller.go

```diff
@@ -866,7 +866,8 @@ func (oc *OperatorController) getOrCreateStoreLimit(storeID uint64, limitType st
 		return nil
 	}
 
-	limit := s.GetStoreLimit()
-	limit.Reset(ratePerSec, limitType)
-	return limit
+	if limit := s.GetStoreLimit(); limit.Cap(limitType) != ratePerSec {
+		oc.cluster.GetBasicCluster().ResetStoreLimit(storeID, limitType, ratePerSec)
+	}
+	return s.GetStoreLimit()
```

Comment:
How about just changing back and not adding a new interface?

Reply:
`ResetStoreLimit` will clone first. So if it doesn't compare the cap, it will waste `clone` once.